woocommerce/woocommerce-ios
# Dashboard: reset the last sync timestamp after new orders

Part of #6966 


### Description


Since the store analytics don't change unless new orders come in, we currently set a minimum time before we make an API request for each tab if the trigger isn't from the first launch and user pull-to-refresh:

- daily: 60 s
- weekly, monthly: 1 h
- yearly: 12 h

In this PR, we want to reset the last sync timestamp when a local or remote order comes in so that it fetches the latest data after the new order(s).

- **Observing remote orders:** `backgroundNotifications` is added to `PushNotesManager` so that `StoreStatsAndTopPerformersViewController` can observe both background and foreground notifications for store orders on the given site. In order to check on the notification site ID, `siteID` is added to `PushNotification` from the notification payload `blog` property
- **Observing local orders:** since Core Data context change notifications have more support in the Yosemite layer, I decided to add a new `OrderAction.observeInsertedOrders` for observing new local orders and also for easier unit testing

Please note that there is another subtask to always refresh the visitor stats in https://github.com/woocommerce/woocommerce-ios/issues/6966.

### Testing instructions


Please test this on a device since push notifications are not supported in the simulator.

- Launch the app, log in if needed
- In the My store tab, tap on a tab that isn't Today so that the minimum sync time is longer than 1 minute for testing
- Go to the Orders tab, and create an order (please note that locally created orders don't always contribute to order stats, there's probably some logic on the backend that excluded local orders under certain conditions)
- Go back to the My store tab, tap on another time range tab and then back to the previously selected tab --> an API request should be made for the previously selected tab because the last sync timestamp is reset by the local order
- Put the app in the background
- On web, place an order on the same site --> a push notification should be sent
- Open the app again
- In the My store tab, tap on another time range tab and then back to the previously selected tab --> an API request should be made for the previously selected tab because the last sync timestamp is reset by the remote order

---
